geo/src/utils.rs

```diff
@@ -53,3 +53,21 @@ where
         }
     }
 }
+
+// The Rust standard library has `max` for `Ord`, but not for `PartialOrd`
+pub fn partial_max<T: PartialOrd>(a: T, b: T) -> T {
```

Comment:
hm... can we use T::min from float traits?

Reply:
oh, I forgot CoordinateType isn't necessarily a Float. 😅

Reply:
yeah...our usage of both `Float` and `CoordinateType` is confusing at times